### PR TITLE
Bug 1854479: enforce permissions on Thanos querier endpoints

### PR DESCRIPTION
**Pull request by simonpasquier**
~~This needs https://github.com/openshift/kube-rbac-proxy/pull/26 first~~

### CODE REVIEW

File: test/e2e/user_workload_monitoring_test.go [hunk 1]
```diff
@@ -724,6 +724,46 @@ func assertTenancyForMetrics(t *testing.T) {
...
+		}
+		defer resp.Body.Close()
+
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode/100 == 2 {
+			return fmt.Errorf("unexpected status code response, got %d (%s)", resp.StatusCode, framework.ClampMax(b))
```
Comment (s-urbaniak) [inline test/e2e/user_workload_monitoring_test.go:759]: nit: can we be a bit more explicit about what we expect? i.e. `expected request to be rejected, but got status code response %d (%s)` ?

Comment (simonpasquier) [reply test/e2e/user_workload_monitoring_test.go]: :+1:

**Pull request merged**